internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -0,0 +1,236 @@
+		}
+		return targetResolverBuilder.Build(target, cc, opts)
+	}
+
+	// When the scheme is 'dns' and target resolution on client is not enabled,
+	// resolution should be handled by the proxy, not the client. Therefore, we
+	// bypass the target resolver and store the unresolved target address.
+	if target.URL.Scheme == "dns" && !targetResolutionEnabled {
+		r.targetAddrs = []resolver.Address{{Addr: target.Endpoint()}}
+	} else {
+		if r.targetResolver, err = targetResolverBuilder.Build(target, &wrappingClientConn{r, "target"}, opts); err != nil {
+			return nil, fmt.Errorf("delegating_resolver: unable to build the resolver for target %v : %v", target, err)
+		}
+	}
+
+	if r.proxyResolver, err = r.proxyURIResolver(opts); err != nil {
+		return nil, err
+	}
+	return r, nil
+}
+
+func (r *delegatingResolver) proxyURIResolver(opts resolver.BuildOptions) (resolver.Resolver, error) {
+	proxyBuilder := resolver.Get("dns")
+	if proxyBuilder == nil {
+		return nil, fmt.Errorf("delegating_resolver: resolver for proxy not found for scheme dns")
+	}
+	host := "dns:///" + r.proxyURL.Host
+	u, err := url.Parse(host)
+	if err != nil {
+		return nil, err
+	}
+
+	proxyTarget := resolver.Target{URL: *u}
+	return proxyBuilder.Build(proxyTarget, &wrappingClientConn{r, "proxy"}, opts)
+}
+
+func (r *delegatingResolver) ResolveNow(o resolver.ResolveNowOptions) {
+	if r.targetResolver != nil {
+		r.targetResolver.ResolveNow(o)
+	}
+	if r.proxyResolver != nil {
+		r.proxyResolver.ResolveNow(o)
+	}
+}
+
+func (r *delegatingResolver) Close() {
+	if r.targetResolver != nil {
+		r.targetResolver.Close()
+	}
+	if r.proxyResolver != nil {
+		r.proxyResolver.Close()
+	}
+}
+
+type keyType string
+
+const proxyConnectAddrKey = keyType("grpc.resolver.delegatingresolver.proxyConnectAddr")
+const userKey = keyType("grpc.resolver.delegatingresolver.user")
+
+// SetConnectAddr returns a copy of the provided resolver.Address with attributes
+// containing address to be sent in connect request to proxy.  It's data should
+// not be mutated after calling SetConnectAddr.
+func SetConnectAddr(resAddr resolver.Address, addr string) resolver.Address {
+	resAddr.Attributes = resAddr.Attributes.WithValue(proxyConnectAddrKey, addr)
+	return resAddr
+}
+
+// ProxyConnectAddr returns the proxy connect address in resolver.Address, or nil
+// if not present. The returned data should not be mutated.
+func ProxyConnectAddr(addr resolver.Address) string {
+	return addr.Attributes.Value(proxyConnectAddrKey).(string)
+}
+
+// SetUser returns a copy of the provided address with attributes containing
+// user info. It's data should not be mutated after calling SetUser.
+func SetUser(addr resolver.Address, user *url.Userinfo) resolver.Address {
+	addr.Attributes = addr.Attributes.WithValue(userKey, user)
+	return addr
+}
+
+// User returns the user info in the resolver.Address, or nil if not present.
+// The returned data should not be mutated.
+func User(addr resolver.Address) *url.Userinfo {
```

Comment:
This will need to go in some other package IIUC.  We are doing this so the transport can read it in order to use the proxy, correct?  It would be improper for the transport to ask this specific resolver for information like this.

Reply:
No, It is so that we can add the auth information to the HTTP CONNECT request. Earlier is used to be done in the internal/transport/proxy.go file itself from the proxyURL. Since we done have the proxyURL in the transport package now, I am sending this information as attribute of the proxy address.

Reply:
Added it to a different `atrributes` package